src/internal/connector/onedrive/collection.go

```diff
@@ -252,22 +300,49 @@ func (oc *Collection) populateItems(ctx context.Context) {
 				itemSize = itemInfo.OneDrive.Size
 			}
 
-			itemReader := lazy.NewLazyReadCloser(func() (io.ReadCloser, error) {
-				progReader, closer := observe.ItemProgress(ctx, itemData, observe.ItemBackupMsg, itemName, itemSize)
-				go closer()
-				return progReader, nil
-			})
+			if isFile {
+				itemReader := lazy.NewLazyReadCloser(func() (io.ReadCloser, error) {
+					progReader, closer := observe.ItemProgress(ctx, itemData, observe.ItemBackupMsg, itemName+DataFileSuffix, itemSize)
+					go closer()
+					return progReader, nil
+				})
+
+				oc.data <- &Item{
+					id:   itemName + DataFileSuffix,
+					data: itemReader,
+					info: itemInfo,
+				}
+			}
+
+			if itemMeta != nil {
+				// TODO(meain): This right now creates two
+				// progressbars for each file, one for :data and one
+				// for :meta. Should we have some way to show it a
+				// different manner?
+				metaFileSuffix := MetaFileSuffix
+				if !isFile {
+					metaFileSuffix = DirMetaFileSuffix
+				}
+				metaReader := lazy.NewLazyReadCloser(func() (io.ReadCloser, error) {
+					progReader, closer := observe.ItemProgress(ctx, itemMeta, observe.ItemBackupMsg, itemName+metaFileSuffix, itemMetaSize)
+					go closer()
+					return progReader, nil
+				})
+
+				oc.data <- &Item{
```

Comment:
As of now, details file contains the names of metadata files. We can drop that, but one advantage of having them there is so that we are also aware of empty directories. Otherwise we might miss restoring empty directories(which is the current behavior). I am assuming the reason behind us not restoring empty directories as of now was we needed files to create a "Collection".

Reply:
>might miss restoring empty directories(which is the current behavior)

To be honest, I'm not sure if we've ever actually pinned down the expected behavior of empty directories as far as whether we should back them up and/or restore them. Right now I think we're pretty inconsistent about their handling between OneDrive and Exchange and even within a service between full backup vs. incremental backup

If we keep them in backup details we'll need to add helper code to filter them out when displaying information to the user. I'm not sure I have an answer as to whether they should/shouldn't be in details to start with though

Reply:
If we end up omitting the metadata file if the values are just the defaults (as per the discussion [here](https://github.com/alcionai/corso/pull/2148#discussion_r1073074066), we will have to keep the metadata file so as to know when to ask kopia for a metadata file.

Reply:
@vkamra any thoughts around what should be going in the details file as well as if we should be restoring empty folders?